magento/magento2
# Fix URL path of Sourcemap(.map) files

### Description (*)


### Related Pull Requests


### Fixed Issues (if relevant)

1. [x] resolves magento/magento2#37290: Fix URL path of Sourcemap(.map) files

### Manual testing scenarios (*)

Verified the problem and fix on Magento 2.4.5-p1 and 2.4-develop. The problem doesn't exist on Magento 2.3.7-p4 so probably one of the recent changes in 2.4.x to the grunt code or one of the updated npm packages causes the problem.

Steps to reproduce:
1. Setup clean Magento
2. Copy `package.json.sample` to `package.json`
3. Run `npm install`
4. Run `cp Gruntfile.js.sample Gruntfile.js`
5. Run `grunt exec:luma`
6. Run `grunt less:luma`
7. Run: `grep -r 'sourceMappingURL' pub/static` and notice this output:
```
$ grep -r 'sourceMappingURL' pub/static
pub/static/frontend/Magento/luma/en_US/css/styles-l.css:/*# sourceMappingURL=pub/static/frontend/Magento/luma/en_US/css/styles-l.css.map */
pub/static/frontend/Magento/luma/en_US/css/styles-m.css:/*# sourceMappingURL=pub/static/frontend/Magento/luma/en_US/css/styles-m.css.map */
```
Which are incorrect paths to the sourcemap files, the browser tries to load these incorrectly resulting in 404 errors and doesn't find the original less files in the inspector:
![Screenshot 2023-01-05 at 13 45 50](https://user-images.githubusercontent.com/85479/210784294-0d805e5a-5618-4ddf-9c1b-0f32e0e83d05.png)

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#37290: Fix URL path of Sourcemap(.map) files
